src/data_table/local_data_table.js

```diff
@@ -459,6 +461,56 @@ class LocalDataTable extends View {
     this.dataTable.on("page", this._bulkCheckboxAdjust);
   }
 
+  _setupPaginationHistory() {
+    this.dataTable.on("page", function() {
+      var page = this.dataTable.fnPagingInfo().iPage;
+      if (page !== this._parseQueryString(window.location.search)) {
+        history.pushState({}, "pagination", this._createSearchString(this.dataTable.fnPagingInfo().iPage));
+      }
+    }.bind(this));
+    window.onpopstate = () => {
+      this._pageToSearchPage();
+    };
+    this.on("render", this._afterRender());
+  }
+
+  _afterRender() {
+    this._pageToSearchPage();
+  }
+
+  _pageToSearchPage() {
+    let pageNumber = this._parseQueryString(window.location.search);
+    if (pageNumber >= 0) {
+      this.page(this._parseQueryString(window.location.search));
+    }
+  }
+
+  _createSearchString(pageNumber) {
+    var queryString = window.location.search;
+    var searchString = "?";
+    pageNumber++;
+    var pageString = "page=" + pageNumber;
+    if (queryString[0] === '?') {
```

Comment:
i think this can all be simplified to read a lot nicer.  its looks like all this is doing is trying to add the page number to the already existing query params.

```
let urlParams = $.deparam(window.location.href.split("?")[1] || "");
urlParams.page = pageNumber;
return "?" + $.param(urlParams)
```